test/integration/testExecute.js

```diff
@@ -163,180 +165,249 @@ describe('Execute test', function ()
+        },
+        {
+          name: 'xml_with_attribute_and_custom_parser',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: '##'
+        },
+        {
+          name: 'xml_skip_attributes',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}>${TEST_XML_VAL}</${TEST_HEADER}></node>`,
+          attributeNamePrefix: '##'
+        },
+        {
+          name: 'xml_attribute_only',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}></${TEST_HEADER}></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: '##'
+
+        },
+        {
+          name: 'xml_attribute_empty_value',
+          type: 'XML',
+          fileExtension: '.xml',
+          sampleData: `<node><${TEST_HEADER} ${TEST_ATTRIBUTE_NAME}=${TEST_ATTRIBUTE_VALUE}/></node>`,
+          ignoreAttributes: false,
+          attributeNamePrefix: '##'
+        }
+      ];
+
+    const rowAsserts = (testCase, row) => {
+      // Check the column, header, and value is correct
+      const attributeWithPrefix = testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME;
+
+      if (testCase.name === 'xml_single_element') {
+        assert.strictEqual(row[TEST_COL][TEST_HEADER], TEST_XML_VAL);
+      } else if (testCase.name === 'xml_single_element_with_attribute') {
+        assert.strictEqual(row[TEST_COL][TEST_HEADER][ELEMENT_VALUE_FIELD], TEST_XML_VAL);
+      } else if (testCase.name === 'xml_with_attribute_and_custom_parser') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], TEST_XML_VAL);
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][attributeWithPrefix], TEST_ATTRIBUTE_VALUE);
+      } else if (testCase.name === 'xml_skip_attributes') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER], TEST_XML_VAL);
+        assert.equal(row[TEST_COL]['node'][TEST_HEADER][testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME], undefined);
+      } else if (testCase.name === 'xml_attribute_only') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], undefined);
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME], TEST_ATTRIBUTE_VALUE);
+      } else if (testCase.name === 'xml_attribute_empty_value') {
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][ELEMENT_VALUE_FIELD], undefined);
+        assert.strictEqual(row[TEST_COL]['node'][TEST_HEADER][testCase.attributeNamePrefix + TEST_ATTRIBUTE_NAME], TEST_ATTRIBUTE_VALUE);
+      }
+    };
+
+    for (let index = 0; index < testCases.length; index++) {
```

Comment:
maybe let's rewrite it to `testCases.forEach`?

Reply:
ok